### PR TITLE
[e2e demo 3/3] API to read data

### PR DESCRIPTION
**Pull request by manolisliolios**
## Description 

Add API to read the data saved from the event indexer.

## Test Plan 

How did you test the new or updated feature?

---
If your changes are not user-facing and do not break anything, you can skip the following section. Otherwise, please briefly describe what has changed under the Release Notes section.

### Type of Change (Check all that apply)

- [ ] protocol change
- [ ] user-visible impact
- [ ] breaking change for a client SDKs
- [ ] breaking change for FNs (FN binary must upgrade)
- [ ] breaking change for validators or node operators (must upgrade binaries)
- [ ] breaking change for on-chain data layout
- [ ] necessitate either a data wipe or data migration

### CODE REVIEW

File: examples/trading/api/README.md [hunk 1]
```diff
@@ -0,0 +1,120 @@
+# Escrow API Demo
+
+This demo is built to showcase how we can build an event indexer + API
+to easily & efficiently serve on-chain data for our app.
+
+The demo indexer uses polling to watch for new events.
+
+Everything is pre-configured on testnet, but can be easily tweaked to work on any other network.
```
Comment (amnn) [inline examples/trading/api/README.md:8]: Can we add a brief explanation of how here?

File: examples/trading/api/README.md [hunk 2]
```diff
@@ -0,0 +1,120 @@
...
+```
+npx ts-node helpers/create-demo-data.ts
+```
+
+3. Produce demo escrows
+```
+npx ts-node helpers/create-demo-escrows.ts
+```
+
+
```
Comment (amnn) [inline examples/trading/api/README.md:60]: nit: (too much) whitespace

File: examples/trading/api/README.md [hunk 3]
```diff
@@ -0,0 +1,120 @@
...
+| Parameter 	| Expected value    	|
+|-----------	|-------------------	|
+| cancelled 	| 'true' \| 'false' 	|
+| swapped   	| 'true' \| 'false' 	|
+| recipient 	| string            	|
+| sender    	| string            	|
+
+Example query: 
+
+`GET /escrows?sort=asc&limit=10&cursor=15&recipient=0x2&cancelled=false`
```
Comment (amnn) [inline examples/trading/api/README.md:105]: Would it be possible to format this as a `curl` request for ease of copying? Also could you accompany this with a short description of what it does?

File: examples/trading/api/README.md
Comment (amnn) [inline examples/trading/api/README.md:1]: ❤️ Love this README.

File: examples/trading/api/server.ts [hunk 4]
```diff
@@ -0,0 +1,88 @@
...
+
+		return res.send(formatPaginatedResponse(escrows));
+	} catch (e) {
+		console.error(e);
+		return res.status(400).send(e);
+	}
+});
+
+app.listen(3000, () =>
+	console.log(`
```
Comment (amnn) [inline examples/trading/api/server.ts:86]: nit: what's with the newline here?

**Review by amnn — APPROVED** [review]
Love it!  Mostly comments about the README.

File: examples/trading/api/server.ts (hunk 4 above)
Comment (manolisliolios) [reply examples/trading/api/server.ts:86]: That comes from the global (monorepo) linter, probably based on a max width setting.

File: examples/trading/api/README.md (hunk 3 above)
Comment (manolisliolios) [reply examples/trading/api/README.md:105]: Absolutely, will do!

File: examples/trading/api/README.md (hunk 1 above)
Comment (manolisliolios) [reply examples/trading/api/README.md:8]: Good point, will do!

File: examples/trading/api/server.ts (hunk 4 above)
Comment (amnn) [reply examples/trading/api/server.ts:86]: I meant within the string here -- I would be really suprised if a linter changed the string output!

**Pull request merged**